### ISSUE TITLE
Adventure is coming to paper - Bungeecord to be deprecated

### ISSUE THREAD

**Issue by JorelAli** [JorelAli/CommandAPI#188]
> Starting with Paper 1.16.5 build 473, Paper now includes Adventure (https://github.com/KyoriPowered/adventure). This means that developers can now use every feature from Adventure in plugins without having to include/shade Adventure as a dependency. This allows you to easily send messages that utilise all chat features of Minecraft (for example, including itemstacks or entities in hover events), and more. It replaces all methods that used the BungeeCord chat API, and adds a bunch of new methods that only accepted strings in the past. You can still use the old methods, however they are marked as @Deprecated (and may be removed in the future), since Adventure replaces them.
You can find the documentation for Adventure here: https://docs.adventure.kyori.net/
For support with Adventure in general: https://discord.gg/MMfhJ8F

The CommandAPI should add support for this.

**Comment by JorelAli** [comment]
Scheduled for v5.10.

Support is now added via the CommandAPI's `config.yml` file. It now has an option `use-paper-adventure`, which when `true`, allows Paper Adventure API support. For example:

```java
@Override
public void onEnable() {

    new CommandAPICommand("ccspigot").withArguments(new ChatArgument("test"))
    .executes((sender, args) -> {
        BaseComponent[] message = (BaseComponent[]) args[0];
        
        //Broadcast the message to everyone on the server
        Bukkit.getServer().spigot().broadcast(message);
    }).register();
    
    new CommandAPICommand("ccpadventure").withArguments(new ChatArgument("test2"))
    .executes((sender, args) -> {
        Component message = (Component) args[0];
        
        //Broadcast the message to everyone on the server
        Bukkit.broadcast(message, "someperm");
    }).register();
}
```
If you use `use-paper-adventure`, then the top command will fail on execution (cast failure, can't convert `TextComponentImpl` to `BaseComponent[]`). If you use the bottom one without `use-paper-adventure`, then the bottom one fails (cast failure, can't convert `BaseComponent[]` to `Component`)

If you use `use-paper-adventure` without having a compatible version of paper (i.e. old paperspigot version), the CommandAPI will dump a warning in the console

**Comment by JorelAli** [comment]
This has been rewritten to not depend on a `config.yml` file. Instead, you can now access the paper-specific methods using `AdventureChatArgument` and `AdventureChatComponentArgument`.

This makes everything much much easier to use and doesn't break compatibility with existing plugins that use the CommandAPI.

**Comment by JorelAli** [comment]
Implemented on v5.10
